stl/inc/xutility

```diff
@@ -208,6 +208,15 @@ struct _Get_rebind_alias<_Ty, _Other, void_t<typename _Ty::template rebind<_Othe
     using type = typename _Ty::template rebind<_Other>;
 };
 
+#if _HAS_CXX23
+template <class _Raw, class _Source>
+_NODISCARD constexpr auto&& _Forward_as(auto&& _Val) noexcept {
+    _STL_INTERNAL_STATIC_ASSERT(same_as<_Source&, decltype(_Val)&>);

```

Comment:
My bad: we'll need `is_same_v` instead of `same_as` for non-Concepts mode:
```suggestion
    _STL_INTERNAL_STATIC_ASSERT(is_same_v<_Source&, decltype(_Val)&>);
```

Reply:
Fun fact, i originally put it into type traits to be close to the common_reference machinery.

But then I thought @CaseyCarter wanted same as and it is throughput optimized so I moved it to xutility